blockchain/fullblocktests/generate.go

```diff
@@ -2133,3 +2181,138 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
+	// the current tip which expects the header to be accepted to the main
+	// chain.
+	//
+	// rejected creates and appends a single rejectHeader test instance for
+	// the current tip.
+	accepted := func() {
+		tests = append(tests, []TestInstance{
+			acceptHeader(g.tipName, g.tip),
+		})
+	}
+	rejected := func(code blockchain.ErrorCode) {
+		tests = append(tests, []TestInstance{
+			rejectHeader(g.tipName, g.tip, code),
+		})
+	}
+	// ---------------------------------------------------------------------
+	// Generate enough blocks to have mature coinbase outputs to work with.
+	//
+	//   genesis -> bm0 -> bm1 -> ... -> bm99
+	// ---------------------------------------------------------------------
+	var testInstances []TestInstance
+	for i := uint16(0); i < coinbaseMaturity; i++ {
+		blockName := fmt.Sprintf("bm%d", i)
+		g.nextBlock(blockName, nil)
+		g.saveTipCoinbaseOut()
+		testInstances = append(testInstances, acceptHeader(g.tipName,
+			g.tip))
+	}
+	tests = append(tests, testInstances)
+
+	// Collect spendable outputs.  This simplifies the code below.
+	var outs []*spendableOut
+	for i := uint16(0); i < coinbaseMaturity; i++ {
+		op := g.oldestCoinbaseOut()
+		outs = append(outs, &op)
+	}
+	// Adding a block to the existing chain and accepting it
+	//
+	// ... -> b0
+	g.nextBlock("b0", nil)
+	accepted()
+	// Create a block with invalid proof of work and reject that header
+	//
+	//	 ... -> b0()
+	// 					\-> b1(1)
+	g.setTip("b0")
+	b1 := g.nextBlock("b1", outs[1])
+	// This can't be done inside a munge function passed to nextBlock
+	// because the block is solved after the function returns and this test
+	// requires an unsolved block.
+	{
+		origHash := b1.BlockHash()
+		for {
+			// Keep incrementing the nonce until the hash treated as
+			// a uint256 is higher than the limit.
+			b1.Header.Nonce++
+			blockHash := b1.BlockHash()
+			hashNum := blockchain.HashToBig(&blockHash)
+			if hashNum.Cmp(g.params.PowLimit) >= 0 {
+				break
+			}
+		}
+		g.updateBlockState("b46", origHash, "b46", b1)
+	}
+	rejected(blockchain.ErrHighHash)
+	// Create block with an invalid proof-of-work limit.
+	//
+	//   ... -> b0()
+	//                 \-> b2(1)
+	g.setTip("b0")
+	g.nextBlock("b2", outs[1], func(b *wire.MsgBlock) {
+		b.Header.Bits--
+	})
+	rejected(blockchain.ErrUnexpectedDifficulty)
+	// Create block with an invalid negative proof-of-work limit.
+	//
+	//   ... -> b0()
+	//                 \-> b3(1)
+	g.setTip("b0")
+	b3 := g.nextBlock("b3", outs[1])
+	// This can't be done inside a munge function passed to nextBlock
+	// because the block is solved after the function returns and this test
+	// involves an unsolvable block.
+	{
+		origHash := b3.BlockHash()
+		b3.Header.Bits = 0x01810000 // -1 in compact form.
+		g.updateBlockState("b3", origHash, "b3", b3)
+	}
+	rejected(blockchain.ErrUnexpectedDifficulty)
+	// Adding a block with valid header
+	//
+	// ... -> b0() -> b4(2)
```

Comment:
Why was b4(1) changed to b4(2)?

Reply:
I changed it earlier, to test immature spend error, but that was not being detected on headers download, so changed it back in the code, forgot to change the comment.